modules/scan/class-admin-bar-notice.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * A class that adds the scan notice to the admin bar.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Scan;
+
+use Automattic\Jetpack\Assets;
+use Automattic\Jetpack\Redirect;
+
+/**
+ * Class Main
+ *
+ * Responsible for loading the admin bar notice if threats are found.
+ *
+ * @package Automattic\Jetpack\Scan
+ */
+class Admin_Bar_Notice {
+	const SCRIPT_NAME    = 'jetpack-scan-show-notice';
+	const SCRIPT_VERSION = '1';
+
+	/**
+	 * The singleton instance of this class.
+	 *
+	 * @var Admin_Bar_Notice
+	 */
+	protected static $instance;
+
+	/**
+	 * Get the singleton instance of the class.
+	 *
+	 * @return Admin_Bar_Notice
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new Admin_Bar_Notice();
+			self::$instance->init_hooks();
+		}
+
+		return self::$instance;
+	}
+	/**
+	 * Initalize the hooks as needed.
+	 */
+	private function init_hooks() {
+		if ( is_multisite() ) {
```

Comment:
Should we bail early when a site does not have the right plan?

Reply:
Since the site capabilities state (does the site have scan) is something that is available on the .com side. 
Which the /scan endpoint uses already. I think creating another cache " that knows about that state and then using would involve more logic and more calls. So this is the best alternative. We should make sure that the /scan endpoint is more efficient.